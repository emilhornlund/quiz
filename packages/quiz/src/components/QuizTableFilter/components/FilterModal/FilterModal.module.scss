@import '../../../../styles/colors';
@import '../../../../styles/helpers';

.filterModalContainer {
  display: flex;
  flex-direction: column;

  @include mobile {
    gap: $mobile-narrow-layout-gutter;
  }

  @include tablet {
    gap: $tablet-narrow-layout-gutter;
  }

  @include desktop {
    gap: $desktop-narrow-layout-gutter;
  }

  .row {
    display: flex;
    flex-direction: column;

    @include mobile {
      row-gap: $mobile-narrow-layout-gutter * 0.5;
    }

    @include tablet {
      row-gap: $tablet-narrow-layout-gutter * 0.5;
    }

    @include desktop {
      row-gap: $desktop-narrow-layout-gutter * 0.5;
    }

    .label {
      display: flex;
      flex-direction: column;
      font-weight: 700;

      @include mobile {
        @include fontSize(14px);
      }

      @include tablet {
        @include fontSize(16px);
      }

      @include desktop {
        @include fontSize(18px);
      }
    }

    &.buttonGroup {
      display: flex;
      flex-direction: row;

      @include mobile {
        column-gap: $mobile-narrow-layout-gutter;
      }

      @include tablet {
        column-gap: $tablet-narrow-layout-gutter;
      }

      @include desktop {
        column-gap: $desktop-narrow-layout-gutter;
      }

      > div {
        flex: 1;
      }
    }
  }
}
