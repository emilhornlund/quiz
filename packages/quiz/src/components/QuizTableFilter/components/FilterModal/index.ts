export type { FilterModalProps } from './FilterModal'
export { default } from './FilterModal'
