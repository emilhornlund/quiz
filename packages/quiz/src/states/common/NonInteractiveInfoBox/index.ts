export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
