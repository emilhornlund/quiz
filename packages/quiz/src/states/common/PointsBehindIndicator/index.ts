export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export { default } from './PointsBehindIndicator'
