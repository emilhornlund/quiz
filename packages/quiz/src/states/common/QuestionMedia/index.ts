export type { QuestionMediaProps } from './QuestionMedia'
export { default } from './QuestionMedia'
