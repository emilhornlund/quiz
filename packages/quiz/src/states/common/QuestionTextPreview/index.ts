export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export { default } from './QuestionTextPreview'
