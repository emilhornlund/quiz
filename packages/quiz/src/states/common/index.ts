export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
