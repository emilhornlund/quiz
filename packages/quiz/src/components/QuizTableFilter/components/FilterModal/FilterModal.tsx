import { GameMode, LanguageCode, QuizVisibility } from '@quiz/common'
import React, { FC, useState } from 'react'

import {
  GameModeLabels,
  LanguageLabels,
  QuizVisibilityLabels,
} from '../../../../models/labels.ts'
import { classNames } from '../../../../utils/helpers.ts'
import Button from '../../../Button'
import Modal from '../../../Modal'
import Select from '../../../Select'

import styles from './FilterModal.module.scss'

export interface FilterOptions {
  visibility?: QuizVisibility
  languageCode?: LanguageCode
  mode?: GameMode
  sort?: 'title' | 'created' | 'updated'
  order?: 'asc' | 'desc'
}

export interface FilterModalProps extends FilterOptions {
  showVisibilityFilter: boolean
  open: boolean
  onClose: () => void
  onApply: (options: FilterOptions) => void
}

const FilterModal: FC<FilterModalProps> = ({
  visibility,
  languageCode,
  mode,
  sort,
  order,
  showVisibilityFilter,
  open,
  onClose,
  onApply,
}) => {
  const [internalFilterOptions, setInternalFilterOptions] =
    useState<FilterOptions>({
      visibility,
      languageCode,
      mode,
      sort,
      order,
    })

  const handleFilterOptionChange = <K extends keyof FilterOptions>(
    key: K,
    value: FilterOptions[K],
  ): void => {
    setInternalFilterOptions((oldFilterOptions) => ({
      ...oldFilterOptions,
      [key]: value,
    }))
  }

  const handleApplyFilter = () => {
    onApply(internalFilterOptions)
  }

  return (
    <Modal title="Refine Your Quiz Search" open={open}>
      <div className={styles.filterModalContainer}>
        Narrow down your search and find the perfect quiz!
        {showVisibilityFilter && (
          <div className={styles.row}>
            <div className={styles.label}>Visibility</div>
            <Select
              id="visibility-select"
              kind="secondary"
              values={[
                { key: 'none', value: 'none', valueLabel: 'None' },
                ...Object.values(QuizVisibility).map((visibility) => ({
                  key: visibility,
                  value: visibility,
                  valueLabel: QuizVisibilityLabels[visibility],
                })),
              ]}
              value={internalFilterOptions.visibility || 'none'}
              onChange={(value) =>
                handleFilterOptionChange(
                  'visibility',
                  value === 'none' ? undefined : (value as QuizVisibility),
                )
              }
            />
          </div>
        )}
        <div className={styles.row}>
          <div className={styles.label}>Language</div>
          <Select
            id="language-select"
            kind="secondary"
            values={[
              { key: 'none', value: 'none', valueLabel: 'None' },
              ...Object.values(LanguageCode).map((languageCode) => ({
                key: languageCode,
                value: languageCode,
                valueLabel: LanguageLabels[languageCode],
              })),
            ]}
            value={internalFilterOptions.languageCode || 'none'}
            onChange={(value) =>
              handleFilterOptionChange(
                'languageCode',
                value === 'none' ? undefined : (value as LanguageCode),
              )
            }
          />
        </div>
        <div className={styles.row}>
          <div className={styles.label}>Game Mode</div>
          <Select
            id="game-mode-select"
            kind="secondary"
            values={[
              { key: 'none', value: 'none', valueLabel: 'None' },
              ...Object.values(GameMode).map((gameMode) => ({
                key: gameMode,
                value: gameMode,
                valueLabel: GameModeLabels[gameMode],
              })),
            ]}
            value={internalFilterOptions.mode || 'none'}
            onChange={(value) =>
              handleFilterOptionChange(
                'mode',
                value === 'none' ? undefined : (value as GameMode),
              )
            }
          />
        </div>
        <div className={styles.row}>
          <div className={styles.label}>Sort By</div>
          <Select
            id="sort-by-select"
            kind="secondary"
            values={[
              { key: 'title', value: 'title', valueLabel: 'Title' },
              { key: 'created', value: 'created', valueLabel: 'Created' },
              { key: 'updated', value: 'updated', valueLabel: 'Updated' },
            ]}
            value={internalFilterOptions.sort || 'created'}
            onChange={(value) =>
              handleFilterOptionChange(
                'sort',
                value as 'title' | 'created' | 'updated',
              )
            }
          />
        </div>
        <div className={styles.row}>
          <div className={styles.label}>Sort Order</div>
          <Select
            id="sort-order-select"
            kind="secondary"
            values={[
              { key: 'asc', value: 'asc', valueLabel: 'Ascending' },
              { key: 'desc', value: 'desc', valueLabel: 'Descending' },
            ]}
            value={internalFilterOptions.order || 'desc'}
            onChange={(value) =>
              handleFilterOptionChange('order', value as 'asc' | 'desc')
            }
          />
        </div>
        <div className={classNames(styles.row, styles.buttonGroup)}>
          <Button
            id="close-button"
            type="button"
            kind="secondary"
            value="Close"
            onClick={onClose}
          />
          <Button
            id="apply-button"
            type="button"
            kind="call-to-action"
            value="Apply"
            onClick={handleApplyFilter}
          />
        </div>
      </div>
    </Modal>
  )
}

export default FilterModal
