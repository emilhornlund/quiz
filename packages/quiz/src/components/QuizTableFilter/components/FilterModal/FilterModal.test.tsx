import { GameMode, LanguageCode, QuizVisibility } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import FilterModal from './FilterModal'

describe('FilterModal', () => {
  it('should render FilterModal', async () => {
    const { container } = render(
      <FilterModal
        visibility={QuizVisibility.Public}
        languageCode={LanguageCode.English}
        mode={GameMode.Classic}
        sort="created"
        order="desc"
        onClose={() => undefined}
        onApply={() => undefined}
        showVisibilityFilter
        open
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
