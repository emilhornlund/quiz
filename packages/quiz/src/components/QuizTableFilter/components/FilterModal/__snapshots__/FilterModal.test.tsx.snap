// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterModal > should render FilterModal 1`] = `
<div>
  <div
    class="floatingOverlay"
    style="position: fixed; overflow: auto; top: 0px; right: 0px; bottom: 0px; left: 0px;"
  >
    <span
      data-floating-ui-focus-guard=""
      data-type="inside"
      role="button"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
      tabindex="0"
    />
    <div
      aria-labelledby="«r2»"
      class="modalContainer sizeNormal"
      data-floating-ui-focusable=""
      id="«r0»"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="header"
      >
        <div
          class="title"
          id="«r2»"
        >
          Refine Your Quiz Search
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class="filterModalContainer"
        >
          Narrow down your search and find the perfect quiz!
          <div
            class="row"
          >
            <div
              class="label"
            >
              Category
            </div>
            <div
              class="inputContainer"
            >
              <div
                class="selectInputContainer selectInputKindSecondary"
              >
                <select
                  data-testid="test-category-select-select"
                  id="category-select"
                  name="category-select"
                >
                  <option
                    value="none"
                  >
                    None
                  </option>
                  <option
                    value="GENERAL_KNOWLEDGE"
                  >
                    General Knowledge
                  </option>
                  <option
                    value="SCIENCE"
                  >
                    Science
                  </option>
                  <option
                    value="HISTORY"
                  >
                    History
                  </option>
                  <option
                    value="GEOGRAPHY"
                  >
                    Geography
                  </option>
                  <option
                    value="ENTERTAINMENT"
                  >
                    Entertainment
                  </option>
                  <option
                    value="SPORTS"
                  >
                    Sports
                  </option>
                  <option
                    value="LITERATURE"
                  >
                    Literature
                  </option>
                  <option
                    value="TECHNOLOGY"
                  >
                    Technology
                  </option>
                  <option
                    value="ART"
                  >
                    Art
                  </option>
                  <option
                    value="MUSIC"
                  >
                    Music
                  </option>
                  <option
                    value="FILM"
                  >
                    Film
                  </option>
                  <option
                    value="FOOD_AND_DRINK"
                  >
                    Food & Drink
                  </option>
                  <option
                    value="POLITICS"
                  >
                    Politics
                  </option>
                  <option
                    value="MYTHOLOGY"
                  >
                    Mythology
                  </option>
                  <option
                    value="NATURE"
                  >
                    Nature
                  </option>
                  <option
                    value="BUSINESS"
                  >
                    Business
                  </option>
                  <option
                    value="HEALTH"
                  >
                    Health
                  </option>
                  <option
                    value="RELIGION"
                  >
                    Religion
                  </option>
                  <option
                    value="SPACE"
                  >
                    Space
                  </option>
                  <option
                    value="MATHEMATICS"
                  >
                    Mathematics
                  </option>
                  <option
                    value="OTHER"
                  >
                    Other
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-down selectIcon"
                  color="black"
                  data-icon="chevron-down"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                >
                  <path
                    d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label"
            >
              Visibility
            </div>
            <div
              class="inputContainer"
            >
              <div
                class="selectInputContainer selectInputKindSecondary"
              >
                <select
                  data-testid="test-visibility-select-select"
                  id="visibility-select"
                  name="visibility-select"
                >
                  <option
                    value="none"
                  >
                    None
                  </option>
                  <option
                    value="PUBLIC"
                  >
                    Public
                  </option>
                  <option
                    value="PRIVATE"
                  >
                    Private
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-down selectIcon"
                  color="black"
                  data-icon="chevron-down"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                >
                  <path
                    d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label"
            >
              Language
            </div>
            <div
              class="inputContainer"
            >
              <div
                class="selectInputContainer selectInputKindSecondary"
              >
                <select
                  data-testid="test-language-select-select"
                  id="language-select"
                  name="language-select"
                >
                  <option
                    value="none"
                  >
                    None
                  </option>
                  <option
                    value="ab"
                  >
                    Abkhaz
                  </option>
                  <option
                    value="aa"
                  >
                    Afar
                  </option>
                  <option
                    value="af"
                  >
                    Afrikaans
                  </option>
                  <option
                    value="ak"
                  >
                    Akan
                  </option>
                  <option
                    value="sq"
                  >
                    Albanian
                  </option>
                  <option
                    value="am"
                  >
                    Amharic
                  </option>
                  <option
                    value="ar"
                  >
                    Arabic
                  </option>
                  <option
                    value="an"
                  >
                    Aragonese
                  </option>
                  <option
                    value="hy"
                  >
                    Armenian
                  </option>
                  <option
                    value="as"
                  >
                    Assamese
                  </option>
                  <option
                    value="av"
                  >
                    Avaric
                  </option>
                  <option
                    value="ae"
                  >
                    Avestan
                  </option>
                  <option
                    value="ay"
                  >
                    Aymara
                  </option>
                  <option
                    value="az"
                  >
                    Azerbaijani
                  </option>
                  <option
                    value="bm"
                  >
                    Bambara
                  </option>
                  <option
                    value="ba"
                  >
                    Bashkir
                  </option>
                  <option
                    value="eu"
                  >
                    Basque
                  </option>
                  <option
                    value="be"
                  >
                    Belarusian
                  </option>
                  <option
                    value="bn"
                  >
                    Bengali
                  </option>
                  <option
                    value="bh"
                  >
                    Bihari
                  </option>
                  <option
                    value="bi"
                  >
                    Bislama
                  </option>
                  <option
                    value="bs"
                  >
                    Bosnian
                  </option>
                  <option
                    value="br"
                  >
                    Breton
                  </option>
                  <option
                    value="bg"
                  >
                    Bulgarian
                  </option>
                  <option
                    value="my"
                  >
                    Burmese
                  </option>
                  <option
                    value="ca"
                  >
                    Catalan
                  </option>
                  <option
                    value="ch"
                  >
                    Chamorro
                  </option>
                  <option
                    value="ce"
                  >
                    Chechen
                  </option>
                  <option
                    value="ny"
                  >
                    Chichewa
                  </option>
                  <option
                    value="zh"
                  >
                    Chinese
                  </option>
                  <option
                    value="cu"
                  >
                    Church Slavonic
                  </option>
                  <option
                    value="cv"
                  >
                    Chuvash
                  </option>
                  <option
                    value="kw"
                  >
                    Cornish
                  </option>
                  <option
                    value="co"
                  >
                    Corsican
                  </option>
                  <option
                    value="cr"
                  >
                    Cree
                  </option>
                  <option
                    value="hr"
                  >
                    Croatian
                  </option>
                  <option
                    value="cs"
                  >
                    Czech
                  </option>
                  <option
                    value="da"
                  >
                    Danish
                  </option>
                  <option
                    value="dv"
                  >
                    Dhivehi
                  </option>
                  <option
                    value="nl"
                  >
                    Dutch
                  </option>
                  <option
                    value="dz"
                  >
                    Dzongkha
                  </option>
                  <option
                    value="en"
                  >
                    English
                  </option>
                  <option
                    value="eo"
                  >
                    Esperanto
                  </option>
                  <option
                    value="et"
                  >
                    Estonian
                  </option>
                  <option
                    value="ee"
                  >
                    Ewe
                  </option>
                  <option
                    value="fo"
                  >
                    Faroese
                  </option>
                  <option
                    value="fj"
                  >
                    Fijian
                  </option>
                  <option
                    value="fi"
                  >
                    Finnish
                  </option>
                  <option
                    value="fr"
                  >
                    French
                  </option>
                  <option
                    value="ff"
                  >
                    Fula
                  </option>
                  <option
                    value="gl"
                  >
                    Galician
                  </option>
                  <option
                    value="lg"
                  >
                    Ganda
                  </option>
                  <option
                    value="ka"
                  >
                    Georgian
                  </option>
                  <option
                    value="de"
                  >
                    German
                  </option>
                  <option
                    value="el"
                  >
                    Greek
                  </option>
                  <option
                    value="gn"
                  >
                    Guarani
                  </option>
                  <option
                    value="gu"
                  >
                    Gujarati
                  </option>
                  <option
                    value="ht"
                  >
                    Haitian
                  </option>
                  <option
                    value="ha"
                  >
                    Hausa
                  </option>
                  <option
                    value="he"
                  >
                    Hebrew
                  </option>
                  <option
                    value="hz"
                  >
                    Herero
                  </option>
                  <option
                    value="hi"
                  >
                    Hindi
                  </option>
                  <option
                    value="ho"
                  >
                    Hiri Motu
                  </option>
                  <option
                    value="hu"
                  >
                    Hungarian
                  </option>
                  <option
                    value="is"
                  >
                    Icelandic
                  </option>
                  <option
                    value="io"
                  >
                    Ido
                  </option>
                  <option
                    value="ig"
                  >
                    Igbo
                  </option>
                  <option
                    value="id"
                  >
                    Indonesian
                  </option>
                  <option
                    value="ia"
                  >
                    Interlingua
                  </option>
                  <option
                    value="ie"
                  >
                    Interlingue
                  </option>
                  <option
                    value="iu"
                  >
                    Inuktitut
                  </option>
                  <option
                    value="ik"
                  >
                    Inupiaq
                  </option>
                  <option
                    value="ga"
                  >
                    Irish
                  </option>
                  <option
                    value="it"
                  >
                    Italian
                  </option>
                  <option
                    value="ja"
                  >
                    Japanese
                  </option>
                  <option
                    value="jv"
                  >
                    Javanese
                  </option>
                  <option
                    value="kl"
                  >
                    Kalaallisut
                  </option>
                  <option
                    value="kn"
                  >
                    Kannada
                  </option>
                  <option
                    value="kr"
                  >
                    Kanuri
                  </option>
                  <option
                    value="ks"
                  >
                    Kashmiri
                  </option>
                  <option
                    value="kk"
                  >
                    Kazakh
                  </option>
                  <option
                    value="km"
                  >
                    Khmer
                  </option>
                  <option
                    value="ki"
                  >
                    Kikuyu
                  </option>
                  <option
                    value="rw"
                  >
                    Kinyarwanda
                  </option>
                  <option
                    value="rn"
                  >
                    Kirundi
                  </option>
                  <option
                    value="kv"
                  >
                    Komi
                  </option>
                  <option
                    value="kg"
                  >
                    Kongo
                  </option>
                  <option
                    value="ko"
                  >
                    Korean
                  </option>
                  <option
                    value="ku"
                  >
                    Kurdish
                  </option>
                  <option
                    value="kj"
                  >
                    Kwanyama
                  </option>
                  <option
                    value="ky"
                  >
                    Kyrgyz
                  </option>
                  <option
                    value="lo"
                  >
                    Lao
                  </option>
                  <option
                    value="la"
                  >
                    Latin
                  </option>
                  <option
                    value="lv"
                  >
                    Latvian
                  </option>
                  <option
                    value="li"
                  >
                    Limburgish
                  </option>
                  <option
                    value="ln"
                  >
                    Lingala
                  </option>
                  <option
                    value="lt"
                  >
                    Lithuanian
                  </option>
                  <option
                    value="lu"
                  >
                    Luba-Katanga
                  </option>
                  <option
                    value="lb"
                  >
                    Luxembourgish
                  </option>
                  <option
                    value="mk"
                  >
                    Macedonian
                  </option>
                  <option
                    value="mg"
                  >
                    Malagasy
                  </option>
                  <option
                    value="ms"
                  >
                    Malay
                  </option>
                  <option
                    value="ml"
                  >
                    Malayalam
                  </option>
                  <option
                    value="mt"
                  >
                    Maltese
                  </option>
                  <option
                    value="gv"
                  >
                    Manx
                  </option>
                  <option
                    value="mr"
                  >
                    Marathi
                  </option>
                  <option
                    value="mh"
                  >
                    Marshallese
                  </option>
                  <option
                    value="mn"
                  >
                    Mongolian
                  </option>
                  <option
                    value="mi"
                  >
                    Maori
                  </option>
                  <option
                    value="na"
                  >
                    Nauru
                  </option>
                  <option
                    value="nv"
                  >
                    Navajo
                  </option>
                  <option
                    value="ng"
                  >
                    Ndonga
                  </option>
                  <option
                    value="ne"
                  >
                    Nepali
                  </option>
                  <option
                    value="nd"
                  >
                    Northern Ndebele
                  </option>
                  <option
                    value="se"
                  >
                    Northern Sami
                  </option>
                  <option
                    value="nb"
                  >
                    Norwegian Bokmål
                  </option>
                  <option
                    value="nn"
                  >
                    Norwegian Nynorsk
                  </option>
                  <option
                    value="ii"
                  >
                    Nuosu
                  </option>
                  <option
                    value="oc"
                  >
                    Occitan
                  </option>
                  <option
                    value="oj"
                  >
                    Ojibwe
                  </option>
                  <option
                    value="or"
                  >
                    Oriya
                  </option>
                  <option
                    value="om"
                  >
                    Oromo
                  </option>
                  <option
                    value="os"
                  >
                    Ossetian
                  </option>
                  <option
                    value="ps"
                  >
                    Pashto
                  </option>
                  <option
                    value="fa"
                  >
                    Persian
                  </option>
                  <option
                    value="pl"
                  >
                    Polish
                  </option>
                  <option
                    value="pt"
                  >
                    Portuguese
                  </option>
                  <option
                    value="pa"
                  >
                    Punjabi
                  </option>
                  <option
                    value="pi"
                  >
                    Pali
                  </option>
                  <option
                    value="qu"
                  >
                    Quechua
                  </option>
                  <option
                    value="ro"
                  >
                    Romanian
                  </option>
                  <option
                    value="rm"
                  >
                    Romansh
                  </option>
                  <option
                    value="ru"
                  >
                    Russian
                  </option>
                  <option
                    value="sm"
                  >
                    Samoan
                  </option>
                  <option
                    value="sg"
                  >
                    Sango
                  </option>
                  <option
                    value="sa"
                  >
                    Sanskrit
                  </option>
                  <option
                    value="sc"
                  >
                    Sardinian
                  </option>
                  <option
                    value="gd"
                  >
                    Scottish Gaelic
                  </option>
                  <option
                    value="sr"
                  >
                    Serbian
                  </option>
                  <option
                    value="sn"
                  >
                    Shona
                  </option>
                  <option
                    value="sd"
                  >
                    Sindhi
                  </option>
                  <option
                    value="si"
                  >
                    Sinhala
                  </option>
                  <option
                    value="sk"
                  >
                    Slovak
                  </option>
                  <option
                    value="sl"
                  >
                    Slovene
                  </option>
                  <option
                    value="so"
                  >
                    Somali
                  </option>
                  <option
                    value="nr"
                  >
                    Southern Ndebele
                  </option>
                  <option
                    value="st"
                  >
                    Southern Sotho
                  </option>
                  <option
                    value="es"
                  >
                    Spanish
                  </option>
                  <option
                    value="su"
                  >
                    Sundanese
                  </option>
                  <option
                    value="sw"
                  >
                    Swahili
                  </option>
                  <option
                    value="ss"
                  >
                    Swati
                  </option>
                  <option
                    value="sv"
                  >
                    Swedish
                  </option>
                  <option
                    value="tl"
                  >
                    Tagalog
                  </option>
                  <option
                    value="ty"
                  >
                    Tahitian
                  </option>
                  <option
                    value="tg"
                  >
                    Tajik
                  </option>
                  <option
                    value="ta"
                  >
                    Tamil
                  </option>
                  <option
                    value="tt"
                  >
                    Tatar
                  </option>
                  <option
                    value="te"
                  >
                    Telugu
                  </option>
                  <option
                    value="th"
                  >
                    Thai
                  </option>
                  <option
                    value="bo"
                  >
                    Tibetan
                  </option>
                  <option
                    value="ti"
                  >
                    Tigrinya
                  </option>
                  <option
                    value="to"
                  >
                    Tongan
                  </option>
                  <option
                    value="ts"
                  >
                    Tsonga
                  </option>
                  <option
                    value="tn"
                  >
                    Tswana
                  </option>
                  <option
                    value="tr"
                  >
                    Turkish
                  </option>
                  <option
                    value="tk"
                  >
                    Turkmen
                  </option>
                  <option
                    value="tw"
                  >
                    Twi
                  </option>
                  <option
                    value="ug"
                  >
                    Uighur
                  </option>
                  <option
                    value="uk"
                  >
                    Ukrainian
                  </option>
                  <option
                    value="ur"
                  >
                    Urdu
                  </option>
                  <option
                    value="uz"
                  >
                    Uzbek
                  </option>
                  <option
                    value="ve"
                  >
                    Venda
                  </option>
                  <option
                    value="vi"
                  >
                    Vietnamese
                  </option>
                  <option
                    value="vo"
                  >
                    Volapük
                  </option>
                  <option
                    value="wa"
                  >
                    Walloon
                  </option>
                  <option
                    value="cy"
                  >
                    Welsh
                  </option>
                  <option
                    value="fy"
                  >
                    West Frisian
                  </option>
                  <option
                    value="wo"
                  >
                    Wolof
                  </option>
                  <option
                    value="xh"
                  >
                    Xhosa
                  </option>
                  <option
                    value="yi"
                  >
                    Yiddish
                  </option>
                  <option
                    value="yo"
                  >
                    Yoruba
                  </option>
                  <option
                    value="za"
                  >
                    Zhuang
                  </option>
                  <option
                    value="zu"
                  >
                    Zulu
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-down selectIcon"
                  color="black"
                  data-icon="chevron-down"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                >
                  <path
                    d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label"
            >
              Game Mode
            </div>
            <div
              class="inputContainer"
            >
              <div
                class="selectInputContainer selectInputKindSecondary"
              >
                <select
                  data-testid="test-game-mode-select-select"
                  id="game-mode-select"
                  name="game-mode-select"
                >
                  <option
                    value="none"
                  >
                    None
                  </option>
                  <option
                    value="CLASSIC"
                  >
                    Classic
                  </option>
                  <option
                    value="ZERO_TO_ONE_HUNDRED"
                  >
                    0 to 100
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-down selectIcon"
                  color="black"
                  data-icon="chevron-down"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                >
                  <path
                    d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label"
            >
              Sort By
            </div>
            <div
              class="inputContainer"
            >
              <div
                class="selectInputContainer selectInputKindSecondary"
              >
                <select
                  data-testid="test-sort-by-select-select"
                  id="sort-by-select"
                  name="sort-by-select"
                >
                  <option
                    value="title"
                  >
                    Title
                  </option>
                  <option
                    value="created"
                  >
                    Created
                  </option>
                  <option
                    value="updated"
                  >
                    Updated
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-down selectIcon"
                  color="black"
                  data-icon="chevron-down"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                >
                  <path
                    d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label"
            >
              Sort Order
            </div>
            <div
              class="inputContainer"
            >
              <div
                class="selectInputContainer selectInputKindSecondary"
              >
                <select
                  data-testid="test-sort-order-select-select"
                  id="sort-order-select"
                  name="sort-order-select"
                >
                  <option
                    value="asc"
                  >
                    Ascending
                  </option>
                  <option
                    value="desc"
                  >
                    Descending
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-down selectIcon"
                  color="black"
                  data-icon="chevron-down"
                  data-prefix="fas"
                  role="img"
                  viewBox="0 0 448 512"
                >
                  <path
                    d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="row buttonGroup"
          >
            <div
              class="buttonInputContainer buttonInputKindSecondary"
            >
              <button
                data-testid="test-close-button-button"
                id="close-button"
                name="close-button"
                type="button"
              >
                <span>
                  Close
                </span>
              </button>
            </div>
            <div
              class="buttonInputContainer buttonInputKindCallToAction"
            >
              <button
                data-testid="test-apply-button-button"
                id="apply-button"
                name="apply-button"
                type="button"
              >
                <span>
                  Apply
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      data-floating-ui-focus-guard=""
      data-type="inside"
      role="button"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
      tabindex="0"
    />
  </div>
</div>
`;
